front/hooks/usePersistedNavigationSelection.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from "react";
+import { useCookies } from "react-cookie";
+
+// client-side counterpart of persisted_navigation_selection.ts
+
+export type NavigationSelectionType = {
+  lastWorkspaceId?: string;
+  lastSpaceId?: string;
+};
+
+export const NAVIGATION_SELECTION_COOKIE_NAME = "navigationSelection";
+
+export const usePersistedNavigationSelection = () => {
+  const [cookies, setCookie] = useCookies([NAVIGATION_SELECTION_COOKIE_NAME]);
```

Comment:
Why not using local storage instead? IIRC, storing preferences in cookies isn't highly recommended.

My sense is that it it does not need to be secure and isn't needed by the SSR/API then local storage should be the go-to solution.

Reply:
Ok update: We actually use it on SSR, might be worth adding a comment here then to explain why cookies over local storage!

Reply:
You have a comment at the top explaining that it has a server-side counterpart :) (but okay no soucy)